src/OpenTelemetry/Internal/SelfDiagnosticsConfigRefresher.cs

```diff
@@ -202,7 +209,7 @@ private void OpenLogFile(string newLogDirectory, int newFileSize)
         try
         {
             Directory.CreateDirectory(newLogDirectory);
-            var fileName = Path.GetFileName(Process.GetCurrentProcess().MainModule.FileName) + "."
+            var fileName = Path.GetFileName(Process.GetCurrentProcess().MainModule?.FileName ?? "OpenTelemetrySdk") + "."
```

Comment:
Why change the file name format?

Reply:
Who knows! According to the nullable decorations from runtime `MainModule` could be null, or `FileName` could be null. This is just making the code defensive just in case.